tests/src/hwtracer_ykpt.rs

```diff
@@ -56,22 +56,13 @@ pub extern "C" fn __hwykpt_assert_basic(trace: *mut Box<dyn Trace>) {
     let mut ykpt_mapper = HWTMapper::new(&mut ykpt_itr);
 
     let mut last = None;
-    let mut ipt_blks = Vec::new();
-    while let Some(blk) = ipt_mapper.next(last) {
-        ipt_blks.push(blk.unwrap());
-        last = ipt_blks.last();
-    }
-
-    let mut ykpt_blks = Vec::new();
-    while let Some(blk) = ykpt_mapper.next(last) {
-        ykpt_blks.push(blk.unwrap());
-        last = ykpt_blks.last();
-    }
-
-    for pair in ipt_blks.iter().zip_longest(ykpt_blks) {
-        match pair {
-            Both(expect, got) => assert_eq!(expect, &got),
-            _ => panic!("iterator lengths differ"),
-        }
+    while let Some(got) = ipt_mapper.next(last.as_ref()) {
+        let got = got.unwrap();
+        let expect = ykpt_mapper.next(last.as_ref()).unwrap().unwrap();
+        dbg!(&expect, &got);
```

Comment:
I don't think the debugging output was supposed to be left in?

Reply:
oops :)

Reply:
Fixed in 00787dc1b9d795980e5ec6e28dc01676910da7ff